test/tests/functional/pbs_qstat_count.py

```diff
@@ -0,0 +1,271 @@
+        """
+        Verify that total_jobs and queued_count reported by the output
+        of qstat -Bf matched the number of jobs submitted, where each
+        job uses ncpus=1.
+        """
+        counts = self.find_state_counts(Running_count)
+        self.assertEqual(counts['total_jobs'],
+                         counts['all_state_count'], 'Job count incorrect')
+        self.assertEqual(counts['expected_queued_count'], counts['Queued'],
+                         'Queued count incorrect')
+
+    def test_queued_no_restart(self):
+        """
+        The test case verifies that the reported queued_count in qstat -Bf
+        without a server restart is equal to the number of jobs
+        (each using ncpus=1) submitted.
+        """
+        jid = []
+        for i in range(4):
+            j = Job(TEST_USER)
+            jid.append(self.server.submit(j))
+
+        a = {ATTR_h: None}
+        j = Job(TEST_USER, a)
+        self.server.submit(j)
+
+        self.submit_waiting_job(600)
+
+        # Wait for jobs to go in R state
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid[1])
+        self.verify_count(2)
+
+    def test_queued_restart(self):
+        """
+        The test case verifies that the queued_count reported by state_count
+        in output of qstat -Bf equals the number of jobs submitted
+        (each using ncpus=1) even after the server is restarted.
+        """
+        jid = []
+        for i in range(4):
+            j = Job(TEST_USER)
+            jid.append(self.server.submit(j))
+
+        a = {ATTR_h: None}
+        j = Job(TEST_USER, a)
+        self.server.submit(j)
+
+        self.submit_waiting_job(600)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid[1])
+
+        self.server.restart()
+        self.verify_count(2)
+
+    def test_queued_no_restart_multiple_queue(self):
+        """
+        The test case verifies that the queued_count reported in the output
+        of qstat -Bf matches the number of jobs submitted in different queues
+        (each using ncpus=1) without server restart.
+        """
+        # create 2 queues
+        qname = ['workq1', 'workq2']
+        for i in range(2):
+            a = {
+                'queue_type': 'Execution',
+                'enabled': 'True',
+                'started': 'True'}
+            self.server.manager(MGR_CMD_CREATE, QUEUE,
+                                a, qname[i], expect=True)
+
+        q1_attr = {ATTR_queue: 'workq1'}
+        q2_attr = {ATTR_queue: 'workq2'}
+
+        # submit jobs, 3 in each queue
+        j = Job(TEST_USER, q1_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        j = Job(TEST_USER, q2_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
```

Comment:
a comment to explain why the submitting of jobs is separated in two parts will help.